packages/cli/src/modules/insights/database/repositories/insights-by-period.repository.ts

```diff
@@ -223,21 +231,23 @@ export class InsightsByPeriodRepository extends Repository<InsightsByPeriod> {
 		const dropTemporaryTable = sql`
 			DROP TABLE rows_to_compact;
 		`;
+		try {
```

Comment:
What do you think about moving the try up, right below the `this.isRunningCompaction = true`? 
In between there are lots of function calls that may fail (`getSql`, `escapeField`, `getAggregationQuery`, etc.) and then leave the class instance in an invalid state where it would never compact again because the finally block never ran.

Reply:
Good idea, done !